src/transformers/generation/utils.py

```diff
@@ -1651,12 +1633,6 @@ def generate(
             )
 
         elif generation_mode == GenerationMode.BEAM_SEARCH:
-            if generation_config.num_return_sequences > generation_config.num_beams:
-                raise ValueError("`num_return_sequences` has to be smaller or equal to `num_beams`.")
-
-            if stopping_criteria.max_length is None:
-                raise ValueError("`max_length` needs to be a stopping_criteria for now.")
```

Comment:
This exception can't be reached: we always set max_length before initializing stopping_criteria, which means we will always have a max_length-related stopping criteria.